packages/ddp-server/writefence.js

```diff
@@ -84,9 +94,16 @@ _.extend(DDPServer._WriteFence.prototype, {
     if (self.fired)
       throw new Error("write fence already activated?");
     if (self.armed && !self.outstanding_writes) {
-      self.fired = true;
-      _.each(self.completion_callbacks, function (f) {f(self);});
-      self.completion_callbacks = [];
+      self.outstanding_writes++;
+      _.each(self.before_fire_callbacks, function (f) {f(self);});
```

Comment:
I wonder if we need to think about what happens if one of these throws?  We aren't for onAllCommitted callbacks, but (a) that's probably a mistake (b) the onBeforeFire callback in practice is more complex than the only existing onAllCommitted callback (eg, it blocks on network activity).